### PR TITLE
Introduce dollar sign escaping

### PR DESCRIPTION
**Pull request by sventschui**
**What**:
Introduces the ability to escape dollar signs using a backslash (`"\\$"`). Using double backslash will escape the backslash (`"\\\\$"`)

e.g
```
"cross-env FOO=\\$BAR echo $FOO" -> $BAR will *not* be replaced
"cross-env FOO=\\\\$BAR echo $FOO -> $BAR will be replaced with env var $BAR
```

**Why**:
I needed to have a dollar sign within my env var value

**How**:
Modifing the variable replacements regex

**Checklist**:

- [x] Documentation
- [x] Tests
- [x] Ready to be merged 
- [x] Added myself to contributors table

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,6 +88,8 @@ the parent. This is quite useful for launching the same command with different
 env variables or when the environment variables are too long to have everything
 in one line.
 
+If you preceed a dollar sign with an odd number of backslashes the expression statement will not be replaced. Note that this means backslashes after the JSON string escaping took place. `"FOO=\\$BAR"` will not be replaced. `"FOO=\\\\$BAR"` will be replaced tough.
```
Comment (kentcdodds) [inline README.md]: > tough

Should be "though"

**Comment by sventschui** [comment]
You're absolutely welcome. This tool saves my ass on a daily basis. Least thing I can do is to contribute ;)

Fixed the typo

File: src/variable.js [hunk 2]
```diff
@@ -40,10 +40,20 @@ function replaceListDelimiters(varValue, varName = '') {
...
+  const envUnixRegex = /(\\*)(\$(\w+)|\${(\w+)})/g // $my_var or ${my_var} or \$my_var
+  return varValue.replace(
+    envUnixRegex,
+    (_, escapeChars, varNameWithDollarSign, varName, altVarName) => {
+      // do not replace things preceded by a odd number of \
+      if (escapeChars.length % 2 === 1) {
+        return varNameWithDollarSign
+      }
+      return (
+        escapeChars.substr(0, escapeChars.length / 2) +
```
Comment (hgwood) [inline src/variable.js]: Could you walk me through why only half the back-slashes are required? I'm not sure I understand.

Comment (sventschui) [reply src/variable.js]: That's what happens in regular js string escaping too. `\\` gets to be a `\`. This is needed to allow somebody to write a `\` in front of a `$VARIABLE` and still replace the variable.

`\$VAR` -> `$VAR`
`\\$VAR` -> `\value`
`\\\$VAR` -> `\\$VAR`
`\\\\$VAR` -> `\\value`

and so on.

Hope this makes it clear

**Comment by sventschui** [comment]
Any news?

**Comment by kentcdodds** [comment]
I think that I'm good with this. Thanks @sventschui :+1:

File: src/variable.js (hunk 2 above)
Comment (hgwood) [reply src/variable.js]: OK. Thank you. What about the other case? If the number of back-slashes is odd, isn't this code going to replace the expression by `$VAR` every time? Shouldn't some back-slashes be retained?

Also I'm not sure your examples are quite consistent. In `\$VAR` -> `$VAR`, the back-slash is escaping the dollar sign. In `\\$VAR` -> `\value` the first back-slash escapes the second and the dollar sign is not escaped. Sounds fine up to there. Then in `\\\$VAR` -> `\\$VAR`, the first black-slash escapes the second, then the third escapes the dollar sign. So the result should be `\$VAR`, shouldn't it? Maybe you made a typo. Fourth example looks right.

**Comment by sventschui** [comment]
Thank you very much!

**Pull request merged**